pandas-stubs/core/groupby/generic.pyi

```diff
@@ -111,7 +112,17 @@ class _DataFrameGroupByNonScalar(DataFrameGroupBy):
 class DataFrameGroupBy(GroupBy):
     def any(self, skipna: bool = ...) -> DataFrame: ...
     def all(self, skipna: bool = ...) -> DataFrame: ...
-    def apply(self, func, *args, **kwargs) -> DataFrame: ...
+    # mypy sees the two overloads as overlapping
+    @overload
+    def apply(  # type: ignore[misc]
+        self, func: Callable[[DataFrame], Series | Scalar], *args, **kwargs
+    ) -> Series: ...
+    @overload
+    def apply(  # type: ignore[misc]
```

Comment:
I replaced my local version of generic.pyi with this one and it seems that neither pyright nor mypy need these two ignores (I would have thought both of them need ignores, a DataFrame is Iterable).

Reply:
If I remove both of these ignores, I get errors from both `pyright` and `mypy`

Reply:
you are right, not sure what I messed up. Unless you add a test covering this second overload, I would be tempted to remove it.

Reply:
> you are right, not sure what I messed up. Unless you add a test covering this second overload, I would be tempted to remove it.

There is a test already.   That's why I needed to add the second overload.

https://github.com/pandas-dev/pandas-stubs/blob/77b357f1b64d3ee39057cedf099f167d02503696/tests/test_frame.py#L558

Reply:
Okay, I'm not a fan of this overload but it seems to work :)